core/src/main/java/hudson/util/io/ReopenableFileOutputStream.java

```diff
@@ -52,8 +55,9 @@ public ReopenableFileOutputStream(File out) {
     private synchronized OutputStream current() throws IOException {
         if (current==null)
             try {
-                current = new FileOutputStream(out,appendOnNextOpen);
-            } catch (FileNotFoundException e) {
+                current = Files.newOutputStream(out.toPath(), StandardOpenOption.CREATE,
+                        appendOnNextOpen ? StandardOpenOption.APPEND : StandardOpenOption.TRUNCATE_EXISTING);
+            } catch (FileNotFoundException | NoSuchFileException e) {
```

Comment:
The [javadoc](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#newOutputStream(java.nio.file.Path,%20java.nio.file.OpenOption...)) for `newOutputStream` does not mandate which of these should be thrown so safer to catch both